### ISSUE TITLE
Call for Developers & Testers (AzPainter + mlib)

### ISSUE THREAD

**Issue by Symbian9** [Symbian9/azpainter#16]
Using this issue I want ask some users of Github (interested in development of image painting/editing software using C language) help test & improve **AzPainter** based on **mlib** toolkit:
* https://github.com/symbian9/azpainter

This help request addressed to next users:
* [ ] Cast @krupunkt
* [ ] Cast @CodeHammers
* [ ] Cast @ferreiradaselva
* [ ] Cast @fcpauldiaz
* [ ] Cast @KrystianTokarz
* [ ] Cast @wtrsltnk
* [ ] Cast @khantilchoksi
* [ ] Cast @vishul
* [ ] Cast @williamshen-nz
* [ ] Cast @aafavelino
* [ ] Cast @wareya
* [ ] Cast @manalmohania
* [ ] Cast @ardaerturk
* [ ] Cast @bpereira615
* [ ] Cast @ZetsubouClown
* ...

> Place an `x` inside the bracket near nickname to confirm this request.

P.S.: If you not interested in AzPainter, then just tell me and your nickname will be removed from this list.

REFERENCE:
* https://github.com/krupunkt/KOU-Paint
* https://github.com/CodeHammers/Paint-For-Kids
* https://github.com/ferreiradaselva/libpixpaint
* https://github.com/fcpauldiaz/TurboPaint
* https://github.com/KrystianTokarz/Graphic-image-editor
* https://github.com/wtrsltnk/imaditor
* https://github.com/khantilchoksi/Visual-Graphics-Editor
* https://github.com/vishul/BMP-image-editor
* https://github.com/williamshen-nz/BMP-Editor
* https://github.com/aafavelino/Editor-de-Imagens-Versao-final
* https://github.com/wareya/warpaint
* https://github.com/manalmohania/bmpedit
* https://github.com/ardaerturk/PPM-Image-Editor
* https://github.com/bpereira615/PhotoEditor
* https://github.com/ZetsubouClown/sprite-thingy

**Comment by wjaguar** [comment]
What is the "mlib toolkit" you mentioned?

**Comment by Symbian9** [comment]
> What is the "mlib toolkit" you mentioned?

@wjaguar, **mlib** is indepedent GUI toolkit designied by author of AzPainter in pure C and distributed under BSD license
* https://github.com/Symbian9/azpainter/tree/master/mlib

**Comment by wjaguar** [comment]
>@wjaguar, mlib is indepedent GUI toolkit designied by author of AzPainter in pure C and distributed under BSD license

I see. GTK+1 reinvented with different names. A dead-end approach, then. :(((

Myself, I chose a different path when all that churn in GTK+ and Qt began. The current version of mtPaint is implemented in a toolkit-agnostic (in principle) macro-language: "V-code".
https://github.com/wjaguar/mtPaint/blob/master/src/mainwindow.c#L4933
https://github.com/wjaguar/mtPaint/blob/master/src/vcode.h

Unfortunately, real life intruded and I didn't have time to write the planned GTK+3 and Qt4 backends in addition to baseline GTK+1&2 one - but even the console backend has proven quite useful, giving full scripting capabilities and console mode essentially for free.
http://www.murga-linux.com/puppy/viewtopic.php?p=794720#794720
http://www.murga-linux.com/puppy/viewtopic.php?p=845408#845408

**Comment by Symbian9** [comment]
> I see. GTK+1 reinvented with different names. A dead-end approach, then. :(((

I think this is not a "bug", its a "feature" of `mlib`. As I know, *Azel* has plans for add suport for Wayland into `mlib` soon
* http://azsky2.html.xdomain.jp/memo/index.html

@wjaguar, so, could you atleast look inside AzPainter code for improve it? Maybe, you could help fix some issues?

I very like mtPaint thanks to PuppyLinux (PuppyLinux installed in dualboot near LinuxMint on my PC), but AzPainter is something like I want when think about combine "GIMP + mtPaint".

P.S.: **mtPaint** much closer to **AzPainterB**, than AzPainter.

**Comment by wjaguar** [comment]
>I think this is not a "bug", its a "feature" of mlib.

No, the too-low level of abstraction *is* a bug, in our time of unpredictable changes to foundation-level libs. The more details you control, the more you have to rewrite when (*not* if) the details are no longer there, or changed beyond all recognition.

And besides, the mass of boilerplate code required to control a GTK-workalike is beyond boring. ;) mtPaint binary became significantly smaller with the transition to V-code.

>As I know, Azel has plans for add suport for Wayland into mlib soon

The best-laid plans of mice and men often go awry. (C)
As evidenced by the fate of my own plans for mtPaint.

>@wjaguar, so, could you atleast look inside AzPainter code for improve it? Maybe, you could help fix some issues?

Maybe I could - but Japanese comments are a showstopper. :( To spend significant time Google-translating them, one has to have a really compelling reason; but one cannot readily assess the worth of the code without seeing what the code is about. A catch-22.

Really, I think that adding a passable English translation to comments is the best patch that could happen to AzPainter at this point. Without it, any call for help is foredoomed.

**Comment by Symbian9** [comment]
> adding a passable English translation to comments is the best patch that could happen to AzPainter at this point.

Yeah, now this is on my TODO list
*

**Comment by Symbian9** [comment]
@izzyaxel, would be you interested in contributing to *AzPainter* development? ;-)
